datatypes/mcs_datatype.h

```diff
@@ -953,7 +953,8 @@ class StoreField
   virtual int store_timestamp(int64_t val) = 0;
   virtual int store_string(const char* str, size_t length) = 0;
   virtual int store_varbinary(const char* str, size_t length) = 0;
-  virtual int store_xlonglong(int64_t val) = 0;
+  virtual int store_longlong(int64_t val) = 0;
+  virtual int store_ulonglong(int64_t val) = 0;
```

Comment:
That is pretty strange interface when int64 is used to store an unsigned value.

Reply:
The reason is the underlying field_>store(longlong nr, bool unsigned_val) takes a signed val and a flag indicating if it should be treated as signed or unsigned. We could have store_ulonglong() take an unsigned, but we'd then have to cast it to signed in the function.